json-rpc/src/views.rs

```diff
@@ -40,3 +49,88 @@ impl From<&[u8]> for BytesView {
         Self(hex::encode(bytes))
     }
 }
+
+#[derive(Serialize)]
+pub struct TransactionView {
+    pub version: u64,
+    pub transaction: TransactionDataView,
+}
+
+#[derive(Serialize)]
+#[serde(tag = "type")]
+pub enum TransactionDataView {
+    #[serde(rename = "blockmetadata")]
+    BlockMetadata { timestamp_usecs: u64 },
+    #[serde(rename = "writeset")]
+    WriteSet {},
+    #[serde(rename = "user")]
+    UserTransaction {
+        sender: String,
+        signature: String,
+        public_key: String,
+        sequence_number: u64,
+        max_gas_amount: u64,
+        gas_unit_price: u64,
+        expiration_time: u64,
+        script_name: String,
+        script_args: ScriptArgsView,
+    },
+}
+
+#[derive(Serialize)]
+#[serde(tag = "type")]
+pub enum ScriptArgsView {
+    #[serde(rename = "peer_to_peer")]
```

Comment:
https://serde.rs/variant-attrs.html#rename_all
camel case?

Reply:
I wanted these to be same as returned by 
```
/// Returns a user friendly mnemonic for the transaction type if the transaction is
/// for a known, white listed, transaction.
pub fn get_transaction_name(code: &[u8]) -> String {
    if code == &ADD_VALIDATOR_TXN[..] {
        return "add_validator_transaction".to_string();
    } else if code == &PEER_TO_PEER_TRANSFER_TXN[..] {
        return "peer_to_peer_transaction".to_string();
    } else if code == &PEER_TO_PEER_TRANSFER_WITH_METADATA_TXN[..] {
        return "peer_to_peer_with_metadata_transaction".to_string();
    } else if code == &CREATE_ACCOUNT_TXN[..] {
        return "create_account_transaction".to_string();
    } else if code == &MINT_TXN[..] {
        return "mint_transaction".to_string();
    } else if code == &REMOVE_VALIDATOR_TXN[..] {
        return "remove_validator_transaction".to_string();
    } else if code == &ROTATE_AUTHENTICATION_KEY_TXN[..] {
        return "rotate_authentication_key_transaction".to_string();
    } else if code == &ROTATE_CONSENSUS_PUBKEY_TXN[..] {
        return "rotate_consensus_pubkey_transaction".to_string();
    }
    "<unknown transaction>".to_string()
}
```

Also, for enums its best to specify JSON tag names so it doesn't changes with Rust source code.